app/services/notify_user.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
```

Comment:
You have added this logic in https://github.com/CircuitVerse/CircuitVerse/pull/3312 PR as well. Let @tachyons add his review to that and we can use that as official logic for handling different `notifications` conditionally.